analytics/lib/counts.py

```diff
@@ -237,33 +247,40 @@ def do_drop_single_stat(property: str) -> None:
 ## DataCollector-level operations ##
 
 def do_pull_by_sql_query(property: str, start_time: datetime, end_time: datetime, query: str,
-                         group_by: Optional[Tuple[models.Model, str]]) -> int:
+                         group_by: Optional[Tuple[models.Model, str]], realm: Optional[Realm] = None) -> int:
     if group_by is None:
         subgroup = 'NULL'
-        group_by_clause  = ''
+        group_by_clause = ''
     else:
         subgroup = '%s.%s' % (group_by[0]._meta.db_table, group_by[1])
         group_by_clause = ', ' + subgroup
 
+    if realm is not None:
+        realm_id = realm.id
+    else:
+        realm_id = ""
```

Comment:
I think it'd be more correct to only pass a `realm_id` argument in the case where `realm is not `None`, by constructing an argument dict and mutating it before passing that to `cursor.execute`.  Otherwise, if `realm is None`, we're passing invalid input.